domainservice/src/test/kotlin/ua/marchenko/artauction/artwork/controller/ArtworkNatsControllerTest.kt

```diff
@@ -0,0 +1,263 @@
+package ua.marchenko.artauction.artwork.controller
+
+import artwork.ArtworkProtoFixture
+import artwork.random
+import artwork.toFullArtwork
+import getRandomString
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import ua.marchenko.artauction.artwork.mapper.toArtworkFullProto
+import ua.marchenko.artauction.artwork.mapper.toArtworkProto
+import ua.marchenko.artauction.artwork.mapper.toCreateArtworkFailureResponseProto
+import ua.marchenko.artauction.artwork.mapper.toFullResponse
+import ua.marchenko.artauction.artwork.mapper.toResponse
+import ua.marchenko.artauction.artwork.model.MongoArtwork
+import ua.marchenko.artauction.artwork.repository.ArtworkRepository
+import ua.marchenko.artauction.common.AbstractBaseNatsControllerTest
+import ua.marchenko.artauction.user.model.MongoUser
+import ua.marchenko.artauction.user.repository.UserRepository
+import ua.marchenko.core.artwork.enums.ArtworkStyle
+import ua.marchenko.core.user.enums.Role
+import ua.marchenko.core.user.exception.UserNotFoundException
+import ua.marchenko.internal.NatsSubject
+import ua.marchenko.internal.commonmodels.Error
+import user.random
+import ua.marchenko.internal.commonmodels.artwork.ArtworkStatus as ArtworkStatusProto
+import ua.marchenko.internal.commonmodels.artwork.Artwork as ArtworkProto
+import ua.marchenko.internal.input.reqreply.artwork.CreateArtworkResponse as CreateArtworkResponseProto
+import ua.marchenko.internal.input.reqreply.artwork.FindArtworkByIdRequest as FindArtworkByIdRequestProto
+import ua.marchenko.internal.input.reqreply.artwork.FindArtworkByIdResponse as FindArtworkByIdResponseProto
+import ua.marchenko.internal.input.reqreply.artwork.FindArtworkFullByIdRequest as FindArtworkFullByIdRequestProto
+import ua.marchenko.internal.input.reqreply.artwork.FindArtworkFullByIdResponse as FindArtworkFullByIdResponseProto
+import ua.marchenko.internal.input.reqreply.artwork.FindAllArtworksFullRequest as FindAllArtworksFullRequestProto
+import ua.marchenko.internal.input.reqreply.artwork.FindAllArtworksFullResponse as FindAllArtworksFullResponseProto
+import ua.marchenko.internal.input.reqreply.artwork.FindAllArtworksRequest as FindAllArtworksRequestProto
+import ua.marchenko.internal.input.reqreply.artwork.FindAllArtworksResponse as FindAllArtworksResponseProto
+
+class ArtworkNatsControllerTest : AbstractBaseNatsControllerTest() {
```

Comment:
It's better to have a separate test class for every tested main class

Imagine, you've never seen this code. How would you check that AddArtworkNatsController is covered by tests?